tests/resources/ankaios.resource

```diff
@@ -173,6 +173,19 @@ the workload "${workload_name}" shall not exist
     &{workload_table}=    table to dict    ${list_result}    WORKLOAD NAME
     Dictionary Should Not Contain Key    ${workload_table}    ${workload_name}
 
+the workload "${workload_name}" shall not exist on agent "${expected_agent_name}" within "${timeout}" seconds
+    @{list_result}=    Create List
+    IF    ${timeout} > 0
+        @{list_result}=    Wait For Workload Removal
+        ...    %{ANK_BIN_DIR}${/}ank get workloads
+        ...    ${workload_name}
+        ...    ${expected_agent_name}
+        ...    ${timeout}
+    END
```

Comment:
I see that we used that pattern more often in this file, but in general if someone uses that keyword and passes timeout 0 then the underlying `ank get workloads` is never executed, but the assertion is green (Should be empty list). Then we have a false-positive stest. I think the other occurrences do not have this problem because they do the reverse check on the list or use the timeout different.

Reply:
@lingnoi, maybe you can comment here. I think we can fix the newly introduced code, but would not do a major refactoring of the robot tests inside this PR.
If we have checks that could give a false PASS when used with an accepted argument, we should fix them to avoid always passing tests.